magenta/music/melody_encoder_decoder.py

```diff
@@ -361,19 +365,23 @@ def events_to_input(self, events, position):
       else:
         melody_event = events[lookback_position]
       index = self.melody_event_to_index(melody_event)
-      input_[i * self.num_melody_events + index] = 1.0
+      input_[(i + 1) * self.num_melody_events + index] = 1.0
+
+    num_lookbacks = len(self._lookback_distances)
 
     # Binary time counter giving the metric location of the *next* note.
     n = position + 1
     for i in range(self._binary_counter_bits):
-      input_[3 * self.num_melody_events + i] = 1.0 if (n / 2 ** i) % 2 else -1.0
+      input_[(num_lookbacks + 1) * self.num_melody_events +
+             i] = 1.0 if (n / 2 ** i) % 2 else -1.0
```

Comment:
At the very least using -1.0 seems inconsistent with all of the other binary features.